dnd5esheets/migrations/versions/623e8558c9ba_add_item_full_text_search_table.py

```diff
@@ -0,0 +1,118 @@
+"""add_item_full_text_search_table
+
+Revision ID: 623e8558c9ba
+Revises: a4d56d02c64b
+Create Date: 2023-08-09 15:45:46.542535
+
+"""
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "623e8558c9ba"
+down_revision = "a4d56d02c64b"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    ddls = [
+        # Create a full text search index for items
+        """
+        CREATE VIRTUAL TABLE item_search_index USING fts5(
+            item_id UNINDEXED,
+            language UNINDEXED,
+            name,
+            description,
+            tokenize=porter
+        )
+        """,
+        # Every time a new item is created, insert a new row into the item_search_index FTS table,
+        # one for the default english name/description, and one per translated name/description
+        """
+        CREATE TRIGGER item_after_insert
+            AFTER INSERT ON item
+        BEGIN
+            INSERT INTO item_search_index (rowid, item_id, language, name, description) VALUES (
+                cast(hex('en') as integer) + new.id,
+                new.id,
+                'en',
+                new.name,
+                json_extract(new.json_data, '$.meta.description')
+            );
+            INSERT INTO item_search_index (rowid, item_id, language, name, description)
+                SELECT
+                    cast(hex(json_each.key) as integer) + new.id,
+                    new.id,
+                    json_each.key,
+                    json_extract(json_each.value, '$.name'),
+                    json_extract(json_each.value, '$.description')
+                FROM json_each(json_extract(new.json_data, '$.meta.translations'));
+        END;
+        """,
+        # Populate the search index with the existing items
```

Comment:
Is the order important between the triggers and the population of the table?

Reply:
Not so much no, I could add the population of the table after the triggers. It's important that I define all of them, but the order of definition isn't important.